provider/testing/resource.go

```diff
@@ -3,134 +3,260 @@ package testing
 	"github.com/georgysavva/scany/pgxscan"
+	"github.com/google/go-cmp/cmp"
 	"github.com/hashicorp/go-hclog"
-	"github.com/hashicorp/hcl/v2/gohcl"
-	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/jackc/pgx/v4/pgxpool"
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
-	"github.com/tmccombs/hcl2json/convert"
 )
 
-type ResourceTestData struct {
+type ResourceTestCase struct {
+	Provider       *provider.Provider
 	Table          *schema.Table
-	Config         interface{}
-	Resources      []string
-	Configure      func(logger hclog.Logger, data interface{}) (schema.ClientMeta, error)
+	Config         string
+	SnapshotsDir   string
 	SkipEmptyJsonB bool
 }
 
-func TestResource(t *testing.T, providerCreator func() *provider.Provider, resource ResourceTestData) {
+var ignoreColumns = map[string]bool{
+	"last_updated": true,
+	"cq_id":        true,
+	"meta":         true,
+}
+
+// IntegrationTest - creates resources using terraform, fetches them to db and compares with expected values
+func TestResource(t *testing.T, resource ResourceTestCase) {
+	t.Parallel()
+	t.Helper()
 	if err := faker.SetRandomMapAndSliceMinSize(1); err != nil {
 		t.Fatal(err)
 	}
 	if err := faker.SetRandomMapAndSliceMaxSize(1); err != nil {
 		t.Fatal(err)
 	}
-	ctx := context.Background()
+
+	// No need for configuration or db connection, get it out of the way first
+	// testTableIdentifiersForProvider(t, resource.Provider)
 
 	pool, err := setupDatabase()
 	if err != nil {
 		t.Fatal(err)
 	}
-	defer pool.Close()
+	ctx := context.Background()
 	conn, err := pool.Acquire(ctx)
 	if err != nil {
 		t.Fatal(err)
 	}
 	defer conn.Release()
-	l := logging.New(hclog.DefaultOptions)
-	migrator := provider.NewTableCreator(l)
-	if err := migrator.CreateTable(ctx, conn, resource.Table, nil); err != nil {
+
+	l := testlog.New(t)
+	l.SetLevel(hclog.Debug)
+	resource.Provider.Logger = l
+	tableCreator := provider.NewTableCreator(l)
+	if err := tableCreator.CreateTable(context.Background(), conn, resource.Table, nil); err != nil {
 		assert.FailNow(t, fmt.Sprintf("failed to create tables %s", resource.Table.Name), err)
 	}
-	// Write configuration as a block and extract it out passing that specific block data as part of the configure provider
-	f := hclwrite.NewFile()
-	f.Body().AppendBlock(gohcl.EncodeAsBlock(resource.Config, "configuration"))
-	data, err := convert.Bytes(f.Bytes(), "config.json", convert.Options{})
-	require.Nil(t, err)
-	hack := map[string]interface{}{}
-	require.Nil(t, json.Unmarshal(data, &hack))
-	data, err = json.Marshal(hack["configuration"].([]interface{})[0])
-	require.Nil(t, err)
 
-	testProvider := providerCreator()
+	if err := deleteTables(conn, resource.Table); err != nil {
+		t.Fatal(err)
+	}
 
-	// No need for configuration or db connection, get it out of the way first
-	testTableIdentifiersForProvider(t, testProvider)
+	if resource.SnapshotsDir != "" {
+		if err = fetch(t, &resource); err != nil {
```

Comment:
it should be executed even if we run mock tests

Reply:
done